### PR TITLE
backend/remote: Support -target on plan and apply

### PR DESCRIPTION
**Pull request by apparentlymart**
This is the local-CLI portion of allowing `-target` to be used with remote operations. `-target` addresses given on local CLI are copied into the API request to create a run, which Terraform Cloud will then in turn pass back into `-target` options for the remote Terraform CLI process.

Old versions of Terraform Cloud/Enterprise that lack support for this new attribute would silently ignore the `target-addrs` attribute and run an untargeted plan, so this also includes some version-sniffing so we can continue to return an error for an older server. The design of `go-tfe`'s versioning API combined with the mocking approach of the remote backend makes it annoyingly messy to unit test the case where `-target` isn't available on the server yet, so as a pragmatic compromise we elected to just test that part manually, with the rationale that the "not supported" situation will go away in the not-too-distant future as new versions of Terraform Cloud and Terraform Enterprise roll out, so that's not a long-term maintenance concern.

---

This also includes some unrelated updates to `backend/remote` to deal with some upstream breaking changes in the [hashicorp/go-tfe](https://github.com/hashicorp/go-tfe/pull/117) library, which now requires use of the server's opaque workspace ID rather than the (organization-name, workspace-name) pair it expected before.

### CODE REVIEW

**Comment by mikenomitch** [comment]
Might be worth adding a different default message when using targetAddrs here - https://github.com/hashicorp/terraform/blob/dcbd835ac370c68962df0435b77cc3263ce7b8bd/backend/remote/backend_plan.go#L222

File: backend/remote/backend_apply.go [hunk 1]
```diff
@@ -102,6 +95,23 @@ func (b *Remote) opApply(stopCtx, cancelCtx context.Context, op *backend.Operati
 		))
 	}
 
+	if len(op.Targets) != 0 {
+		currentAPIVersion, err := version.NewVersion(b.client.RemoteAPIVersion())
```
Comment (apparentlymart) [inline backend/remote/backend_apply.go:99]: @radditude maybe we should include a comment here noting that `RemoteAPIVersion` returns an invalid version number (empty string) for versions before 2.3, to help a future reader understand why a parse error is treated the same as a version mismatch below...

Comment (radditude) [reply backend/remote/backend_apply.go]: Good idea! Added 👍

File: backend/remote/backend_common.go [hunk 2]
```diff
@@ -316,6 +316,10 @@ func (b *Remote) costEstimate(stopCtx, cancelCtx context.Context, op *backend.Op
 				b.CLI.Output(b.Colorize().Color("Waiting for cost estimate to complete..." + elapsed + "\n"))
 			}
 			continue
+		case "skipped_due_to_targeting": // TEMP: not available in the go-tfe library yet; will update this to be tfe.CostEstimateSkippedDueToTargeting once that's available.
```
Comment (apparentlymart) [inline backend/remote/backend_common.go:319]: We're planning to submit this `tfe.CostEstimateSkippedDueToTargeting` upstream to `go-tfe` asynchronously from getting this PR in. Will update this to use the named constant in a subsequent PR once it's included in a `go-tfe` release.

**Pull request merged**